src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/OldAndBrokenServiceLocatorAnalyzer.cs

```diff
@@ -155,13 +166,13 @@ AllowedTypeList allowedTypeList
 
 		private static bool IdentifierIsOfDisallowedType(
 			SemanticModel model,
-			ImmutableArray<INamedTypeSymbol> disallowedTypes,
+			ImmutableHashSet<ITypeSymbol> disallowedTypes,
 			IdentifierNameSyntax syntax,
 			CancellationToken cancellationToken
 		) {
-			var actualType = model.GetTypeInfo( syntax, cancellationToken ).Type as INamedTypeSymbol;

```

Comment:
Where the heat was coming from before:

![image](https://user-images.githubusercontent.com/12101120/151671957-5f3242a2-bd8b-46c3-a495-c5cc1a101ef4.png)


![image](https://user-images.githubusercontent.com/12101120/151671945-b3d01663-e07e-434b-9f4b-60acd7503104.png)